packages/core/src/button/button.element.ts

```diff
@@ -100,75 +81,55 @@ export class CdsButton extends CdsBaseButton {
    * - `success`: disables the button and shows a check mark inside the button; auto-triggers to change back to DEFAULT state after 1000 ms
    * - `error`: shows the content of the button (in the context of application, this state is usually entered from a LOADING state. the application should show appropriate error message)
    */
-  @property({ type: String })
-  loadingState: keyof typeof ClrLoadingState = ClrLoadingState.default;
-
-  constructor() {
-    super();
-    ClarityIcons.addIcons(errorStandardIcon, checkIcon);
+  @property({ type: String }) get loadingState() {
+    return this._loadingState;
   }
 
-  firstUpdated(props: Map<string, any>) {
-    super.firstUpdated(props);
-
-    // Find and wrap any text nodes into span elements
-    spanWrapper(this.childNodes);
-
-    if (this.loadingState !== ClrLoadingState.default) {
-      this.updateLoadingState();
+  set loadingState(value: keyof typeof ClrLoadingState) {
+    // track prior disabled state to set prior value after button is re-enabled from a loading state
+    if (this._loadingState === ClrLoadingState.default) {
+      this._disabled = this.disabled;
```

Comment:
The `_disabled` property was the key fix. To preserve the disabled state while not breaking or causing side effects we need to grab what the current disabled state is when updating from `default` to something else. This way when we go back to `default` we can use what the prior `disabled` state was.